docs/src/modules/components/CustomizationPlayground.tsx

```diff
@@ -0,0 +1,335 @@
+  gap: 0.5,
+  width: '250px',
+  padding: theme.spacing(2),
+}));
+
+const NavItemsWrapper = styled(Box)(() => ({
+  display: 'flex',
+  gap: 1,
+  flexWrap: 'wrap',
+}));
+
+const NavLabel = styled(Typography)(({ theme }) => ({
+  marginTop: theme.spacing(2),
+  marginBottom: theme.spacing(1),
+  fontWeight: theme.typography.fontWeightBold,
+  fontSize: theme.typography.pxToRem(12),
+  textTransform: 'uppercase',
+  letterSpacing: '.08rem',
+}));
+
+const NavItemLabel = styled(Typography)(({ theme }) => ({
+  ...theme.typography.caption,
+  letterSpacing: '.08rem',
+  textTransform: 'uppercase',
+  color: theme.palette.text.secondary,
+  fontSize: theme.typography.pxToRem(11),
+  fontweight: 600,
+}));
+
+const NavItem = styled(Button)(({ theme }) => {
+  return {
+    width: 'fit-content',
+    borderRadius: theme.spacing(2),
+    textTransform: 'none',
+    '&:hover, &.hovered': {
+      color: theme.palette.primary.main,
+    },
+  };
+});
+
+const RECOMMENDATION_MESSAGES: { [k in 'warning' | 'success']: string } = {
+  warning:
+    'This might not be the best styling approach for the selected component. You can check out the other options for better results.',
+  success: 'This is the recommended styling approach for the selected component.',
+};
+
+function StylingRecommendation({ type = 'success' }: { type?: 'warning' | 'success' }) {
+  return (
+    <Tooltip title={RECOMMENDATION_MESSAGES[type]}>
+      {type === 'warning' ? (
+        <Chip color="warning" label="Warning" />
+      ) : (
+        <Chip color="success" label="Recommended" />
+      )}
+    </Tooltip>
+  );
+}
+
+const StyledToggleButtonGroup = styled(ToggleButtonGroup)(({ theme }) => ({
+  '& .MuiToggleButtonGroup-grouped': {
+    margin: theme.spacing(0.5),
+    border: 0,
+    '&.Mui-disabled': {
+      border: 0,
+    },
+    '&:not(:first-of-type)': {
+      borderRadius: theme.shape.borderRadius,
+    },
+    '&:first-of-type': {
+      borderRadius: theme.shape.borderRadius,
+    },
+  },
+}));
+
+interface Props extends ToggleButtonProps {
+  selectedColor: ColorKey;
+}
+
+const StyledToggleButton = styled(ToggleButton, {
+  shouldForwardProp: (prop) => prop !== 'selectedColor',
+})(({ theme, selectedColor = 'deepPurple' }: Props & { theme: Theme }) => ({
+  height: theme.spacing(3),
+  minWidth: theme.spacing(3),
+  color: theme.palette.primary.contrastText,
+  background: DEFAULT_COLORS[selectedColor][500],
+  '&:hover': {
+    background: DEFAULT_COLORS[selectedColor][600],
+  },
+  '&.Mui-selected': {
+    background: DEFAULT_COLORS[selectedColor][500],
+    color: theme.palette.primary.contrastText,
+    '&:hover': {
+      background: DEFAULT_COLORS[selectedColor][600],
+    },
+  },
+}));
+
+function ColorSwitcher({
+  handleTokenChange,
+  selectedColor,
+}: {
+  handleTokenChange: (token: 'color', value: keyof typeof DEFAULT_COLORS) => void;
+  selectedColor: ColorKey;
+}) {
+  return (
+    <React.Fragment>
+      <NavItemLabel>Color</NavItemLabel>
```

Comment:
This crashes:

https://github.com/mui/mui-x/assets/3165635/fa59dfdf-a2cf-491f-9728-ce1c5976f67f